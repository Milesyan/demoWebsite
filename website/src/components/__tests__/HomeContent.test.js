/* eslint-disable no-undef */ /*eslint-disable react/react-in-jsx-scope*/

import { HomeContent } from '../index';

const setup = overrideProps => {
  const initProps = {
  };
  const props = { ...initProps, ...overrideProps};
  const wrapper = shallow(
    <HomeContent {...props}/>
  );
  return { wrapper, props };
}

describe('HomeContent snapshots', () => {

  it('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when there is a dialog', () => {
    const { wrapper } = setup();
    wrapper.setState({
      showDialog: true
    });
    expect(wrapper).toMatchSnapshot();
  });

});

describe('HomeContent functions', () => {

  it('should set showDialog to true when request an invite Button is clicked', () => {
    const { wrapper } = setup();
    wrapper.find('Button').simulate('click');
    expect(wrapper.state().showDialog).toBe(true);
  });

  it('should set showDialog to false when showDialog was true and dialog onClose is triggered', () => {
    const { wrapper } = setup();
    wrapper.setState({showDialog: true});
    expect(wrapper.state().showDialog).toBe(true);
    wrapper.find('Connect(RequestInviteDialog)').simulate('close');
    expect(wrapper.state().showDialog).toBe(false);
  });
  
});