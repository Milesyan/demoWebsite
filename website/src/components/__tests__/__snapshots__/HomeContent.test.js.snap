// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContent snapshots should match snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    A Better Way to enjoy every day.
  </h1>
  <div
    className="content"
  >
    Be the first to know when we launch.
  </div>
  <Button
    onClick={[Function]}
  >
    Request an invite
  </Button>
</div>
`;

exports[`HomeContent snapshots should match snapshot when there is a dialog 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    A Better Way to enjoy every day.
  </h1>
  <div
    className="content"
  >
    Be the first to know when we launch.
  </div>
  <Button
    onClick={[Function]}
  >
    Request an invite
  </Button>
  <Connect(RequestInviteDialog)
    onClose={[Function]}
  />
</div>
`;
